frontend/src/pages/InvitePage/InvitePage.tsx

```diff
@@ -0,0 +1,169 @@
+    // const searchParams = new URLSearchParams(searchFragment);
+    return searchParams.get('capabilityUri') || undefined;
+  } catch (e) {
+    return undefined;
+  }
+};
+
+/**
+ * Renders the main component that navigates through the invite workflow.
+ */
+const InvitePage = () => {
+  const navigate = useNavigate();
+  const translate = useTranslate();
+  const capabilityUri = getCapabilityUri(window.location)!;
+  const [showSuccess, setShowSuccess] = useState(false);
+  const [showProviderSelect, setShowProviderSelect] = useState(false as false | 'login' | 'signup');
+  const [inviterProfile, setInviterProfile] = useState(null as null | Record<string, unknown>);
+  const [capability, setCapability] = useState(null as null | Record<string, unknown>);
+  const notify = useNotify();
+  const { data: identity } = useGetIdentity();
+  const ownProfile = identity?.profileData;
+
+  const outbox = useOutbox();
+
+  if (!capabilityUri) {
+    notify('app.notification.invite_cap_invalid');
+    navigate('/');
+  }
+
+  // Fetch the capability document of the invitation.
+  useEffect(() => {
+    fetch(capabilityUri, { headers: { Accept: 'application/ld+json' } })
+      .then(async r => (r.ok ? r : Promise.reject(r.statusText)))
+      .then(async r => r.json())
+      .catch((error: Error | string) => {
+        notify(translate('app.notification.invite_cap_fetch_error', { error: error.toString() }));
+        navigate('/');
+        return error;
+      })
+      .then(cap => {
+        setCapability(cap);
+      });
+  }, [capabilityUri]);
+
+  // Fetch the inviter profile, once the capability document (with the inviter's profile URI) is available.
+  useEffect(() => {
+    if (capability) {
+      fetch(capability['acl:AccessTo'] as string, {
+        headers: { Accept: 'application/ld+json', Authorization: `Capability ${capabilityUri}` }
+      })
+        .then(async response => response.json())
+        .then(profile => {
+          setInviterProfile(profile);
+        });
+    }
+  }, [capability]);
+
+  // There are multiple page options depending on the current state:
+  // 1. Logged out => show invite page
+  // 1.2 clicked on signup / login => show Provider select
+  // 2. Logged in => show connect
+  // 3. sent connection request => success page
+
+  if (showProviderSelect) {
+    return (
+      <ChoosePodProviderPage
+        isSignup={showProviderSelect === 'signup'}
+        capabilityUri={capabilityUri}
+        onCancel={() => {
+          setShowProviderSelect(false);
+        }}
+      />
+    );
+  }
+  if (showSuccess) {
+    return <InvitePageSuccess inviterProfile={inviterProfile} ownProfile={ownProfile} />;
+  }
+
+  // Logged in and inviter profile fetched.
+  if (identity?.id && inviterProfile && ownProfile) {
+    const onConnectClick = () => {
+      outbox
+        .post({
+          '@context': 'https://activitypods.org/context.json',
+          type: 'Offer',
```

Comment:
The `@semapps/activitypub-components` package has an `ACTIVITY_TYPES` object to avoid typos.